chat/conversations.ts

```diff
@@ -33,6 +33,28 @@ import isChannel = Interfaces.isChannel;
  */
 const CONVERSATION_CACHE_UPDATE_FREQ_IN_MS = 1000;
 
+
+function shouldEnableLogging(
```

Comment:
Considering that every method call to this is just this bit in the middle of the message parsing
```ts
if shouldEnableLogging() {
         core.logs.logMessage();
    }

```

I would have just made this a member function of the abstract conversation class that also calls the logMessage() method after passing the checks, with the logic to check if the message itself is an ad worked in as well.

Could just be something like 
```ts
function logMessage(message: Message) {
    if(settings.shouldLog) {
    // etc.

    core.logs.logMessage(...)
    }
}
```